src/plugins/event-plugins/__tests__/TTIPlugin.test.ts

```diff
@@ -101,3 +115,108 @@ describe('Time to Interactive - Plugin Tests', () => {
+            document,
+            'prerendering'
+        );
+
+        try {
+            // Define document.prerendering as true
+            Object.defineProperty(document, 'prerendering', {
+                configurable: true,
+                value: true
+            });
+
+            const plugin: TTIPlugin = new TTIPlugin();
+
+            plugin.load(context);
+
+            const prerenderingChangeEvent = new Event('prerenderingchange');
+            document.dispatchEvent(prerenderingChangeEvent);
+
+            expect(plugin).toBeDefined();
+        } finally {
+            if (originalDescriptor) {
+                Object.defineProperty(
+                    document,
+                    'prerendering',
+                    originalDescriptor
+                );
+            } else {
+                delete (document as any).prerendering;
+            }
+        }
+    });
+
+    test('Plugin handles missing Performance API gracefully', async () => {
+        const originalPerformance = global.performance;
+
+        try {
+            global.performance = {
+                ...originalPerformance,
+                getEntriesByType: undefined
+            } as any;
+
+            const plugin: TTIPlugin = new TTIPlugin();
+
+            plugin.load(context);
+
+            expect(plugin).toBeDefined();
+        } finally {
+            global.performance = originalPerformance;
+        }
+    });
+
+    test('Plugin handles Performance API errors gracefully', async () => {
+        // Mock performance.getEntriesByType to throw an error
+        global.performance.getEntriesByType = jest
+            .fn()
+            .mockImplementation(() => {
+                throw new Error('Test error');
+            });
+
+        const plugin: TTIPlugin = new TTIPlugin();
+
+        expect(() => plugin.load(context)).not.toThrow();
+    });
+
+    test('Plugin records TTI event on normal page load', async () => {
+        global.performance.getEntriesByType = jest.fn().mockReturnValue([
+            {
+                activationStart: 0 // Zero value indicates normal page load
+            }
+        ]);
+
+        const plugin: TTIPlugin = new TTIPlugin();
+
+        plugin.load(context);
+
+        await new Promise((resolve) => process.nextTick(resolve));
+
+        expect(record).toHaveBeenCalledWith(TIME_TO_INTERACTIVE_EVENT_TYPE, {
+            value: 201,
+            version: '1.0.0'
+        });
+    });
+
+    test('Plugin records TTI event for prerendered page', async () => {
+        global.performance.getEntriesByType = jest.fn().mockReturnValue([
+            {
+                activationStart: 100 // Non-zero value indicates prerendering
+            }
+        ]);
+
+        const plugin: TTIPlugin = new TTIPlugin();
+
+        plugin.load(context);
+
+        await new Promise((resolve) => process.nextTick(resolve));
+        expect(record).toHaveBeenCalledWith(TIME_TO_INTERACTIVE_EVENT_TYPE, {
+            value: 201,
```

Comment:
question: why is the value both `201` here and above? shouldn't they be different (e.g. this one be 101=201-100)